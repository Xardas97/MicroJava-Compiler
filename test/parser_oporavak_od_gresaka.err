ERROR 22:02:22,703 - Lose ime promenljive ili parametra na liniji: 2
ERROR 22:02:22,705 - Losa deklaracija konstante na liniji: 3
ERROR 22:02:22,706 - Losa deklaracija konstanti na liniji: 4
ERROR 22:02:22,709 - Lose ime promenljive ili parametra na liniji: 6
ERROR 22:02:22,712 - Lose ime promenljive ili parametra na liniji: 7
ERROR 22:02:22,712 - Iskaz ne moze biti prepoznat na liniji: 11
ERROR 22:02:22,716 - Los logicki izraz na liniji: 14
ERROR 22:02:22,729 - Linija 12: Identifikator c ne postoji
ERROR 22:02:22,729 - Linija 12: Dodela vrednosti nije moguca, tipovi nisu kompatibilni
ERROR 22:02:22,730 - Parsiranje NIJE uspesno zavrseno!
