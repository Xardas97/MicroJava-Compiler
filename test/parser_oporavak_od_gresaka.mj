program parser_oporavak_od_gresaka
    int a = 3, b;
    const bool c, d = false;
    const + = 3;
{
    void method1(int a a, int b) {}
    void method2(int c++) {}

    void main()
    {
        b = +;
        c = true;
        
        if (a > b || c && && true || d) {
            return;
        }
    }
}