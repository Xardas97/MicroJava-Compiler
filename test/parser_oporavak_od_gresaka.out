INFO  22:02:22,727 - Linija 2: Nadjeno Var b: Int, 0, 0
INFO  22:02:22,727 - Linija 3: Nadjeno Con d: Bool, 0, 0
INFO  22:02:22,727 - Linija 6: Nadjeno Meth method1: None, 0, 0
INFO  22:02:22,727 - Linija 6: Nadjeno Var b: Int, 0, 1
INFO  22:02:22,727 - Linija 7: Nadjeno Meth method2: None, 0, 0
INFO  22:02:22,729 - Linija 9: Nadjeno Meth main: None, 0, 0
INFO  22:02:22,729 - Linija 12: Koriscenje argumenta Var noObj: None, -1, -1
INFO  22:02:22,729 - ==============SADRZAJ TABELE SIMBOLA===================
INFO  22:02:22,730 - 
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: bool, -1, -1 
Prog parser_oporavak_od_gresaka: notype, 0, 1 
   Var b: int, 0, 0 
   Con d: bool, 0, 0 
   Meth method1: notype, 0, 2 
      Var b: int, 0, 1 

   Meth method2: notype, 0, 1 

   Meth main: notype, 0, 0 



